src/stepper/stepper.ts

```diff
@@ -60,6 +60,19 @@ function isIrreducible(node: substituterNodes, context: Context) {
   )
 }
 
+function isStatementsReducible(progs: es.Program, context: Context): boolean {
+  if (progs.body.length === 0) return false
+  if (progs.body.length > 1) return true
+
+  const [lastStatement] = progs.body
+
+  if (lastStatement.type === 'ExpressionStatement') {
+    return !isIrreducible(lastStatement.expression, context)
+  } else {
+    return true
+  }
```

Comment:
Let's use early return to avoid "arrowhead"-style code:

```suggestion
  if (lastStatement.type !== 'ExpressionStatement') {
    return true
  }
  return !isIrreducible(lastStatement.expression, context)
```

Reply:
By the way, why does it return a negation of the reducibility of the last expression?

Reply:
it should return if the expression is reducible, but since the function that exists is in the negation form, it is double negated to achieve isReducibleExpression

Reply:
Oh, right, my bad, misread the function name, thanks!